packages/scheduler/src/Scheduler.js

```diff
@@ -177,7 +177,7 @@ function workLoop(hasTimeRemaining, initialTime) {
       break;
     }
     const callback = currentTask.callback;
-    if (callback !== null) {
+    if (typeof callback === 'function') {
```

Comment:
I think for now we should change this to:

```suggestion
    if (callback != null) {
```

More complex checks should be done when the task is scheduled, as @gaearon said.

Reply:
The `typeof` check might counterintuitively be faster because it doesn't need to check for the weird `document.all` case that `!=` takes into account. In the past `!=` was known to deopt things. Not sure it's still the case but I think the `typeof` check is fine.

Reply:
Oh, nice!